hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java

```diff
@@ -51,11 +63,31 @@ void shouldCallHxInsightClientGetAnswer()
     {
         // given
         String questionId = "questionId";
+        AnswerResponse hXAnswer = new AnswerResponse();
+        hXAnswer.setQuestionId(questionId);
+        hXAnswer.setAnswer("Some answer");
+        given(mockHxInsightClient.getAnswer(questionId)).willReturn(hXAnswer);
 
         // when
-        objectUnderTest.readAll(questionId, null);
+        CollectionWithPagingInfo<AnswerModel> answerResponse = objectUnderTest.readAll(questionId, mockParameters);
 
         // then
         then(mockHxInsightClient).should().getAnswer(questionId);
+        Collection<AnswerModel> answerResponseEntries = answerResponse.getCollection();
+        assertEquals(1, answerResponseEntries.size());
+        AnswerModel answer = answerResponseEntries.iterator().next();
+        assertEquals(hXAnswer.getAnswer(), answer.getAnswer());
+        assertEquals(hXAnswer.getQuestionId(), answer.getQuestionId());
```

Comment:
If you construct an expected object then I think you can do a single assertion here:
```
assertEquals(expected, answerResponseEntries);
```

Reply:
Good point - changed.